torch/utils/data/datapipes/iter/callable.py

```diff
@@ -56,10 +58,28 @@ def __init__(self,
         self.fn = fn  # type: ignore[assignment]
         self.args = () if fn_args is None else fn_args
         self.kwargs = {} if fn_kwargs is None else fn_kwargs
+        if nesting_level < -1:
+            raise Exception("nesting_level must be -1 or >= 0.")
+        self.nesting_level = nesting_level
+
+    def _apply(self, data, nesting_level, fn, args, kwargs):
+        if nesting_level == 0:
+            return fn(data, *args, **kwargs)
+        elif nesting_level > 0:
+            if not isinstance(data, list):
+                raise Exception("nesting_level exceeds data pipe depth.")
```

Comment:
IMO, it should be `IndexError`. And, the message seems not right because there is no data pipe depth. Recommend something like `"data nesting_level ({}) out of range".format(self.nesting_level)`